hadoop-hdds/client/src/main/java/org/apache/hadoop/hdds/scm/storage/BlockInputStream.java

```diff
@@ -115,8 +116,13 @@ public BlockInputStream(BlockID blockId, long blockLen, Pipeline pipeline,
    */
   public synchronized void initialize() throws IOException {
 
-    List<ChunkInfo> chunks = getChunkInfos();
+    // Pre-check that the stream has not been intialized already
+    if (initialized) {
+      return;
+    }
+    Preconditions.checkArgument(chunkOffsets == null);
```

Comment:
Do we need this Precondition check?

Reply:
Just extra precaution so that if new methods are implemented then the stream is not initialized twice. It is not necessary for now though. Removed it.